dot/parachain/collator-protocol/validator_side.go

```diff
@@ -115,12 +124,306 @@ func (cpvs CollatorProtocolValidatorSide) handleNetworkEvents(event network.Netw
 	}
 }
 
-func (cpvs CollatorProtocolValidatorSide) ProcessActiveLeavesUpdateSignal() {
-	// NOTE: nothing to do here
+func (cpvs *CollatorProtocolValidatorSide) ProcessActiveLeavesUpdateSignal(
+	signal parachaintypes.ActiveLeavesUpdateSignal) {
+	// I might need to separate the collator protocol into two parts, one that deals with the
+	// network and other that deals with other subsystems.
+	// Make everythin less messing.
+	// https://github.com/paritytech/polkadot-sdk/blob/1b5f4243d159fbb7cf7067241aca8a37f3dbf7ed/polkadot/node/network/bridge/src/rx/mod.rs#L798
+
+	// this active leaves are handled in bridge in rust, read the code of bridge properly
+
+	// TODO update cpvs.activeLeaves by adding new active leaves and removing deactivated ones
+
+	// TODO: get the value for majorSyncing for syncing package
+	// majorSyncing means you are 5 blocks behind the tip of the chain and thus more aggressively
+	// download blocks etc to reach the tip of the chain faster.
+	var majorSyncing bool
+
+	liveHeads := cpvs.liveHeads[:]
```

Comment:
what are you trying to do here? deep copy the slice?

Reply:
looks like a mistake. I could have used cpvs.liveHeads directly. Push the change in latest commit